tests/unit/lms/services/email_preferences_test.py

```diff
@@ -76,10 +77,80 @@ def test_h_userid_if_token_is_invalid_or_expired(
         with pytest.raises(InvalidTokenError):
             svc.h_userid("https://example.com?token=test_token")
 
+    @pytest.mark.parametrize(
+        "existing_preferences,expected_preferences",
+        [
+            (
+                {
+                    "instructor_email_digests.days.1": True,
+                    "instructor_email_digests.days.2": False,
+                    "instructor_email_digests.days.4": False,
+                    "instructor_email_digests.days.6": False,
+                    "instructor_email_digests.days.7": True,
+                    "other_preference": 42,
+                },
+                {
+                    "instructor_email_digests.days.1": True,
+                    "instructor_email_digests.days.2": False,
+                    "instructor_email_digests.days.3": True,
+                    "instructor_email_digests.days.4": False,
+                    "instructor_email_digests.days.5": True,
+                    "instructor_email_digests.days.6": False,
+                    "instructor_email_digests.days.7": True,
+                },
```

Comment:
In this test case `"instructor_email_digests.days.3"` and `"instructor_email_digests.days.5"` are missing from the existing data in the DB, so the service injects `True` values for them. The service also filters out `"other_preference"`.